### ISSUE TITLE
cloudwatch_without_retention_period_specified

### ISSUE THREAD

**Issue by frediana** [Checkmarx/kics#4153]
### Steps to Reproduce the Problem

1.  Create a terraform-aws cloudwatch log group with a retention period using a variable:
```
resource "aws_cloudwatch_log_group" "foo" {
  name              = "foo"
  retention_in_days = var.common.default_log_retention_period
  kms_key_id        = var.kms_key_arn
}
```
2.  When running Kics an error (MEDIUM level) is raised while it should not:
```
CloudWatch Without Retention Period Specified, Severity: MEDIUM, Results: 4
Description: AWS CloudWatch Log groups should have retention days specified
Platform: Terraform
```

The bug is obviously in https://github.com/Checkmarx/kics/blob/master/assets/queries/terraform/aws/cloudwatch_without_retention_period_specified/query.rego

Look like the check is based on an hard-coded list of values.

I would be happy to fix this, but I don't know how to handle this case (the retention_period value should be evaluated).

Another solution could be to only check existence of the 'retention_period', but not sure this is good enough?
### Specifications

-   Version: 1.4.2
-   Platform: Docker

**Comment by felipe-avelar** [comment]
@frediana Hi, I hope you are doing well.
First of all, thanks for reaching us. I see you are using terraform variable to define your value, are you using a `.auto.tfvars` files or `terraform.tfvars` file to define your variable? Today KICS only supports this kind of files as you can see [here](https://github.com/Checkmarx/kics/blob/master/docs/platforms.md#terraform).

If you are using supported files and if possible, could you provide more information on how you are defining common variable?

**Comment by frediana** [comment]
Hello, thanks for your response.

Here is how I'm defining the variable, as a simple variable, with a default value that can/coud be overrident specifying .auto.tfvars or terraform.tfvars  as you suggested. But I have to admit, in my project the default value is always being used.

```
variable "default_log_retention_period" {
  default              =  60
}

resource "aws_cloudwatch_log_group" "foo" {
  name              = "foo"
  retention_in_days = var.default_log_retention_period
  kms_key_id        = var.kms_key_arn
}
```

**Comment by felipe-avelar** [comment]
@frediana Hello again, I made some tests using your example and I couldn't reproduce using `retention_in_days = var.default_log_retention_period`. 🙁

Also, I realized in your first example, you uses  `var.common.default_log_retention_period` instead of `var.default_log_retention_period`, so I tried the following snippet:
```hcl
variable "common" {
  default = {
    default_log_retention_period = 60
  }
}

resource "aws_cloudwatch_log_group" "foo" {
  name              = "foo"
  retention_in_days = var.common.default_log_retention_period
  kms_key_id        = var.kms_key_arn
}
```
KICS didn't complain in this case too. 😕
What I think that might be happening in this case is that you are defining the variable `variable "default_log_retention_period"` and trying to access it as `var.common.default_log_retention_period` instead `var.default_log_retention_period`. If that is not the case and if it is possible, could you provide more information, please?

**Comment by frediana** [comment]
Hello, @felipe-avelar and thank your for the response. I can reproduce the issue with the following situation:

Sorry, I didn't simplified the actual use case I have wich is this precisely:

```
module "common" {
     # have an output called default_log_retention_period with a default value set to 60.
}

resource "aws_cloudwatch_log_group" "foo" {
  name              = "foo"
  retention_in_days = var.common.default_log_retention_period
  kms_key_id        = var.kms_key_arn
}

```

Not that in my case the common variable is actually the output of another module, called common. With this pattern, running kiks 1.4.2 fails this way:

```
CloudWatch Without Retention Period Specified, Severity: MEDIUM, Results: 4
Description: AWS CloudWatch Log groups should have retention days specified
Platform: Terraform

	[1]: ../../path/modules/batch_launcher/log.tf:3
		002:   name              = "/aws/lambda/batch-launcher${var.common.suffix}"
		003:   retention_in_days = var.common.default_log_retention_period
		004:   kms_key_id        = var.kms_key_arn


	[2]: ../../path/modules/compute_unit/log.tf:7
		006:   name              = local.log_group_name
		007:   retention_in_days = var.common.default_log_retention_period
		008:   kms_key_id        = var.kms_key_arn


	[3]: ../../path/modules/engine/log.tf:3=
		002:   name              = "/aws/lambda/${local.lambda_function_name}"
		003:   retention_in_days = var.common.default_log_retention_period
		004:   kms_key_id        = var.kms_key_arn


	[4]: ../../path/modules/monitoring/log.tf:3
		002:   name              = "/aws/lambda/${local.lambda_function_name}"
		003:   retention_in_days = var.common.default_log_retention_period
		004:   kms_key_id        = var.kms_key_arn
```

**Comment by felipe-avelar** [comment]
Hi, @frediana I see your problem now. Right now KICS does not support any kind of modules (in our roadmap we have a task to support verified terraform modules from terraform registry, but not personal modules), also we still do not support output values.

I'll take this topic to our team as soon as possible, to discuss the possibility to include both supports on our roadmap. Also, if you want, you can request this features as a separate issue and can close this one if you want.

If not, as soon as we have a position about this, we will answer you.

**Comment by frediana** [comment]
Hi @felipe-avelar thank your for your time. I shall soon create a separate issue for this feature. For now I just disabled this rule to get my think work.
